src/main/java/com/flowingcode/vaadin/addons/errorwindow/ErrorWindowFactory.java

```diff
@@ -37,6 +37,17 @@ public interface ErrorWindowFactory {
    * @return true if the application is in production mode, false otherwise
    */
   default boolean isProductionMode() {
-    return ("true".equals(System.getProperty("productionMode")));
+    String errorWindowProductionMode = System.getProperty("errorWindowProductionMode");
+    if (errorWindowProductionMode != null) {
+      return Boolean.valueOf(errorWindowProductionMode);
+    }
+
+    String productionMode = System.getProperty("productionMode");
+    if (productionMode != null) {
+      return Boolean.valueOf(productionMode);
+    }
+
+    return VaadinServiceInitListenerImpl.getProductionMode();
```

Comment:
@coderabbitai Should it be `static volatile`? (getProductionMode is always called after productionMode assignment in VaadinServiceInitListenerImpl, but it may be called from a different thread)

Reply:
@coderabbitai productionMode will only be assigned once (and it will not be read before assignment)